File: ByteIoFramer/Api/src/main/java/org/elkoserver/foundation/net/MessageReceiver.java
package org.elkoserver.foundation.net;

/**
 * Interface for the object that a <code>Connection</code> uses to accept incoming
 * messages from the net.
 */
public interface MessageReceiver {
    /**
     * Receive a message from elsewhere.
     *
     * @param message  The message to be received.
     */
    void receiveMsg(Object message);
}

File: ByteIoFramer/Api/src/main/kotlin/org/elkoserver/foundation/net/MessageReceiver.kt
package org.elkoserver.foundation.net

/**
 * Interface for the object that a `Connection` uses to accept incoming
 * messages from the net.
 */
interface MessageReceiver {
    /**
     * Receive a message from elsewhere.
     *
     * @param message  The message to be received.
     */
    fun receiveMsg(message: Any)
}
